message_app/db/message-app.sql

```diff
@@ -24,7 +24,7 @@ CREATE TABLE conversation_request (
     id INTEGER PRIMARY KEY,
     initiator_id INTEGER NOT NULL,
     receiver_id INTEGER NOT NULL,
-    accepted INTEGER NOT NULL, -- 1 means accepted; 0 means has not been accepted or has been declined
+    accepted INTEGER NOT NULL, -- 1 means accepted; 0 means has not been accepted or declined; 2 means has been declined
```

Comment:
If there are more than 2 values, then we should consider turning it to TEXT. But we will write SQLAlchemy code to do that, not directly. The `message-app.sql` should still be maintained for reference though.

Reply:
if we do that, how can we know which 0 means declined and which 0 means neither accepted nor declined when retrieving data?

Reply:
I mean to change them to TEXT as `accepted`, `pending`, and `declined`, and the name of the column could be changed to `status`

Reply:
if we do not change message-app.sql, then the table in the column in the database is still integer

Reply:
I don't quite get your point here. the table in the column?

Reply:
we used message-app.sql to generate database, then we write sqlalchemy model for each table in the database(each model maps with one table). Now we want to change data type of column "accepted" in conversation_request table. So I think we need to change both message-app.sql and sqlalchemy code.

Reply:
wait, doesn't using SQLAlchemy automatically create all the tables for us? @khuolo01

Reply:
yes, it creates the database in case it does not find any database in the path that we provide in config. But if the database exists, it will use that database. In our case, we create the database using message-app.sql at the beginning

Reply:
You can either leave it as-is or make some change to generate the database merely from SQLAlchemy, and keep message-app.sql there for our reference. I will accept whatever you choose.

Reply:
I think we should let SQLAlchemy generate the table for us, and message-app.sql is just for reference. We don't have any important data rn, so we can still delete the old database and create a new one using SQLAlchemy.